### ISSUE TITLE
get_via_lf() doesn't handle empty cells.

### ISSUE THREAD

**Issue by davidorme** [jennybc/googlesheets#94]
I've got a worksheet with empty cells (and which I can't get via ```get_via_csv()``` because the worksheet doesn't provide a link to that interface). When I load it via ```get_via_lf()```, then I get a warning:

```
> dat <- get_via_lf(ss=sheet2, ws=1)
Accessing worksheet titled "Form Responses"
Warning message:
In rbind(entry = c("9/8/2014 21:13:00", "Dennis Ting Teck Wah",  :
  number of columns of result is not a multiple of vector length (arg 1)
> 
```

What is happening is that the individual row entries extracted from the list feed interface have ```NULL``` values for empty cells. When each row passes through ```plyr::llply(function(x) x[var_names] %>% unlist)```, the ```NULL`` values are dropped:

```
> unlist(list(a=1, b=2, c=3, d=NULL, e=5))
a b c e 
1 2 3 5 
> 
```

With empty cells in a worksheet, the output of the ```llply``` function is a list of vectors that includes truncated vectors containing only non-empty cells. These truncated vectors lose alignment with the column titles and the data in them is recycled to match up to the column width.

The XML for empty cells from the list feed, read in by ```gsheets_GET``` looks like this:

```
 <gsx:introductiontotheproject>Above average</gsx:introductiontotheproject>
 <gsx:commentsoptional_2/>
 <gsx:methodsexposition>Average</gsx:methodsexposition>
 <gsx:commentsoptional_3/>
 <gsx:quantityofworkdone>Average</gsx:quantityofworkdone>
 <gsx:commentsoptional_4/>
 <gsx:qualityofdata>Above Average</gsx:qualityofdata>
 ```

This comes out of ```xmlToList``` as ```NULL```, so it isn't obvious to me that this can be trapped earlier (without changing the ```XML``` package or the google list feed syntax). So, I think the best solution is to convert those ```NULL``` values to ```NA``` in ```get_via_lf()``` before turning the list into a vector.

I think I've got a fix and have put in a pull request for it. Sorry to flood the issue tracker - the package functionality is really useful for me!

David

**Comment by jennybc** [comment]
Thanks I will work with your PR.

**Comment by davidorme** [comment]
Hi @jennybc,

Just to confirm that the fixes to #92 and #94 work for me. Old sheets now give the prompt to try ```get_via_lf()```, which now recognises empty cells.

Many thanks,
David

**Comment by jennybc** [comment]
I think we may yet be able to activate `get_via_csv()` for old sheets but that takes some work BTW. That might still happen. It would be different than for new sheets, so not sure it we have same speed advantage. We'll see.
